test/test_ops.py

```diff
@@ -334,12 +337,62 @@ def test_out(self, device, dtype, op):
         sample = samples[0]
         # call it normally to get the expected result
         expected = op(*sample.input, *sample.args, **sample.kwargs)
+        return expected, sample
+
+    @ops(op_db)
+    def test_out(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
         # call it with out=... and check we get the expected result
         out_kwargs = sample.kwargs.copy()
         out_kwargs['out'] = out = torch.empty_like(expected)
         op(*sample.input, *sample.args, **out_kwargs)
-        self.assertEqual(expected, out)
+        self.assertEqual(expected, out, exact_device=True)
+
+    @ops(op_db)
+    def test_out_empty_tensor(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
+        # call it with out=... and check we get the expected result
+        out_kwargs = sample.kwargs.copy()
+        out_kwargs['out'] = out = torch.empty(0,0, dtype=expected.dtype, device=expected.device)
+        op(*sample.input, *sample.args, **out_kwargs)
+        # Verify if the empty out tensor is resized and restrided to whatever
+        # striding is natural for the op to produce
+        self.assertEqual(expected, out, exact_device=True)
 
+    @ops(op_db)
+    def test_out_different_shape(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
+        # call it with out=... and check we get the expected result
+        out_kwargs = sample.kwargs.copy()
+        # Create an empty tensor of different shape than
+        # the expected tensor
+        x,y = expected.shape[0], expected.shape[1]
+        out_kwargs['out'] = out = torch.empty((x+1,y+1), dtype=expected.dtype, device=expected.device)
+        # Verify that using a tensor shape of incorrrect size to out=
+        # raises userwarning.
+        self.assertWarnsRegex(UserWarning,
+                            'An output with one or more elements was resized',
+                            lambda: op(*sample.input, *sample.args, **out_kwargs))
+        # Verify if the out tensor is resized and restrided to whatever
+        # striding is natural for the op to produce
+        self.assertEqual(expected, out, exact_device=True)
+
+    @ops(op_db)
+    def test_out_inplace(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
+        *inp, = sample.input
```

Comment:
Style nit: I think this can just be `inp = sample.input`.

Reply:
I did try doing this. But it did not work :(